src/software/jetson_nano/ansible/playbooks/remote_flash.yml

```diff
@@ -0,0 +1,77 @@
+---
+
+- name: Remote flashing
+  hosts: THUNDERBOTS_HOSTS
+  vars:
+   all_services:
+     - thunderloop
+     - display
+     - announcement
+     - redis
+   all_actions:
+     - stop
+     - sync
+     - start
+   binary_paths:
+     thunderloop: ../../../../../../../../jetson_nano/thunderloop_main
+     announcement: ../../../../../../../../jetson_nano/broadcasts
+     redis: ../../../../../../../../jetson_nano/redis
+     display: ../../../../../../../../jetson_nano/display/robot_ui
+
+  tasks:
+
+    - debug:
+        msg: "[Robot ID = {{ inventory_hostname }}]"
+      tags: always
+
+    - name: extract services & actions from tags
+      set_fact:
+        services: "{{ all_services if ansible_run_tags | intersect(all_services)|length == 0 else ansible_run_tags | intersect(all_services) }}"
+        actions: "{{ all_actions if ansible_run_tags|intersect(all_actions)|length == 0 else ansible_run_tags | intersect(all_actions)}}"
+      tags: always
+
+    - debug:
+        msg: "Performing the following actions: {{ actions }}
+        on the following services: {{ services }}"
+      tags: always
+
+    - name: Stop Services
+      become: true
+      become_method: sudo
+      ansible.builtin.systemd:
+        name: "{{ service_name }}"
+        masked: no
+        daemon_reload: yes
+        state: stopped
+      with_items: "{{ services }}"
+      loop_control:
+        loop_var: service_name
+      when: "'stop' in actions"
+      tags: always
```

Comment:
can we move this after sync binaryies so the order is 

sync 
stop
start

I've had the sync crash on me and then the announcement service be stopped

Reply:
hm i thought if we did sync->stop, sync would fail because it cant overwrite a running binary

Reply:
ah yeah thats true

can we do everything else first, and then
then do announcements last? a bad deploy is taking out announcements lol

Reply:
discussed offline, added an error handling step